config/webpack-metalsmith-connect/index.js

```diff
@@ -28,28 +28,38 @@ function getEntryPoints(buildOptions) {
 }
 
 function compileAssets(buildOptions) {
+  let compileMiddleware = null;
+  let convertPathsMiddleware = null;
+
   return (files, metalsmith, done) => {
-    const convertPathsMiddleware = convertPathsToRelative(buildOptions);
-    const apps = getEntryPoints(buildOptions);
-    const webpackConfig = generateWebpackConfig(buildOptions, apps);
-    const webpackMiddleware = webpackPlugin(webpackConfig);
+    if (!compileMiddleware) {
+      const apps = getEntryPoints(buildOptions);
+      const webpackConfig = generateWebpackConfig(buildOptions, apps);
+      compileMiddleware = webpackPlugin(webpackConfig);
+      convertPathsMiddleware = convertPathsToRelative(buildOptions);
+    }
 
-    webpackMiddleware(files, metalsmith, (err) => {
+    compileMiddleware(files, metalsmith, (err) => {
       if (err) throw err;
       convertPathsMiddleware(files, metalsmith, done);
     });
   };
 }
```

Comment:
The reason this is in here in the first place is that we pass information from Metalsmith to Webpack via `buildOptions`, and some of that information is applied during previous middleware. If we don't initialize inside of the middleware function, then certain properties won't exist yet.